File: spring-kafka-demo/src/main/java/com/example/demo/domain/Review.java
package com.example.demo.domain;

import com.mongodb.DBObject;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Document(collection = "reviews")
public class Review {
    @Id
    private String id;
    private String message;
    private Integer service;
    private Integer comfort;
    private Integer price;
    private Integer distanceFromAirport;
    private Long hotelId;

    public Review() {
    }

    public Review(DBObject dbReview) {
        this.hotelId = (long) dbReview.get("hotelId");
        this.message = (String) dbReview.get("message");
        this.comfort = (int) dbReview.get("comfort");
        this.service = (int) dbReview.get("service");
        this.price = (int) dbReview.get("price");
        this.distanceFromAirport = (int) dbReview.get("distance_from_airport");
    }

    public Integer getDistanceFromAirport() {
        return distanceFromAirport;
    }

    public void setDistanceFromAirport(Integer distanceFromAirport) {
        this.distanceFromAirport = distanceFromAirport;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public Integer getService() {
        return service;
    }

    public void setService(Integer service) {
        this.service = service;
    }

    public Integer getComfort() {
        return comfort;
    }

    public void setComfort(Integer comfort) {
        this.comfort = comfort;
    }

    public Integer getPrice() {
        return price;
    }

    public void setPrice(Integer price) {
        this.price = price;
    }

    public void setHotelId(Long hotelId) {
        this.hotelId = hotelId;
    }

    public Long getHotelId() {
        return hotelId;
    }

    @Override
    public String toString() {
        return "{\"message\":\"" + message + '\"' +
                ", \"service\":" + service +
                ", \"comfort\":" + comfort +
                ", \"price\":" + price +
                ", \"distanceFromAirport\":" + distanceFromAirport +
                ", \"hotelId\":" + hotelId + "}";
    }

}

File: spark-streaming-demo/src/main/scala/com/internal_ring/spark/Review.scala
package com.internal_ring.spark

import org.apache.spark.sql.Row
import org.bson.Document

class Review(var hotel_id: Long, var message: String, var service: Byte, var comfort: Byte, var price: Byte, var distance_from_airport: Byte) {
  def this(row: Row){
    this(
      row.getAs[Long]("hotelId"),
      row.getAs[String]("message"),
      row.getAs[Long]("service").toByte,
      row.getAs[Long]("comfort").toByte,
      row.getAs[Long]("price").toByte,
      row.getAs[Long]("distanceFromAirport").toByte
    )
  }

  def toDocument: Document = {
    new Document()
      .append("hotelId", hotel_id)
      .append("message", message)
      .append("service", service)
      .append("comfort", comfort)
      .append("price", price)
      .append("distanceFromAirport", distance_from_airport)
  }
}
